### PR TITLE
garage: 0.8.2 -> 0.8.3

### PR DESCRIPTION
**Pull request by adamcstephens**
## Description of changes

https://git.deuxfleurs.fr/Deuxfleurs/garage/releases/tag/v0.8.3

https://garagehq.deuxfleurs.fr/documentation/operations/upgrading/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by emilylange** [comment]
Technically unreleated to this bump, but pretty important nonetheless:
Could you please fix the typo in https://github.com/NixOS/nixpkgs/blob/812ccee781267a7bcf8cbd04268e1664b31a5cd6/pkgs/tools/filesystems/garage/default.nix#L60

so ofborg runs the VM tests without manually writing

@ofborg test garage

```patch
diff --git a/pkgs/tools/filesystems/garage/default.nix b/pkgs/tools/filesystems/garage/default.nix
index b6dbe2e89846..df47d8194377 100644
--- a/pkgs/tools/filesystems/garage/default.nix
+++ b/pkgs/tools/filesystems/garage/default.nix
@@ -57,7 +57,7 @@ let
       "sqlite"
     ]);
 
-    passthru = nixosTests.garage;
+    passthru.tests = nixosTests.garage;
 
     meta = {
       description = "S3-compatible object store for small self-hosted geo-distributed deployments";
```

Thanks!

**Comment by adamcstephens** [comment]
> so ofborg runs the VM tests without manually writing

Yes, definitely. Added another commit with this fix.

---

Result of `nixpkgs-review pr 252142` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>garage (garage_0_8 ,garage_0_8_2)</li>
  </ul>
</details>

-----

Result of `nixpkgs-review pr 252142` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>garage (garage_0_8 ,garage_0_8_2)</li>
  </ul>
</details>

-----

Result of `nixpkgs-review pr 252142` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>garage (garage_0_8 ,garage_0_8_2)</li>
  </ul>
</details>

-----

Result of `nixpkgs-review pr 252142` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>garage (garage_0_8 ,garage_0_8_2)</li>
  </ul>
</details>

File: pkgs/tools/filesystems/garage/default.nix [hunk 1]
```diff
@@ -84,9 +84,9 @@ in
...
     garage_0_8_2 = generic {
-      version = "0.8.2";
-      sha256 = "sha256-IlDWbNWI1yXvPPF3HIqQvo79M2FQCtoX1wRLJrDbd9k=";
-      cargoSha256 = "sha256-6l4tDBMcOvckTkEO05rman4hHlmVbBt1nCeX5/dETKk=";
+      version = "0.8.3";
+      sha256 = "sha256-NxkFj76L+LpCWzOWbnN3zdhw9Q16uzPibDs+C+voM/0=";
+      cargoSha256 = "sha256-hbBuUjdlw//s6d24dPBu3R/BTJvmOW1B7tSIXNxLXlU=";
     };
 
     garage_0_8 = garage_0_8_2;
```
Comment (0x4A6F) [inline pkgs/tools/filesystems/garage/default.nix:92]:
```diff
-    garage_0_8_2 = generic {
+    garage_0_8_3 = generic {
-      version = "0.8.2";
-      sha256 = "sha256-IlDWbNWI1yXvPPF3HIqQvo79M2FQCtoX1wRLJrDbd9k=";
-      cargoSha256 = "sha256-6l4tDBMcOvckTkEO05rman4hHlmVbBt1nCeX5/dETKk=";
+      version = "0.8.3";
+      sha256 = "sha256-NxkFj76L+LpCWzOWbnN3zdhw9Q16uzPibDs+C+voM/0=";
+      cargoSha256 = "sha256-hbBuUjdlw//s6d24dPBu3R/BTJvmOW1B7tSIXNxLXlU=";
    };

-    garage_0_8 = garage_0_8_2;
+    garage_0_8 = garage_0_8_3;
```

or add a separate `garage_0_8_3` definition.

You could also look up an [old version bump](https://github.com/NixOS/nixpkgs/commit/12ebd64d985e674ec55e3d7312c5ecea2c692db2) for this package for inspiration (there is also `all-packages.nix` involved):

**Review by teutat3s — APPROVED** [review]

**Review by 0x4A6F — CHANGES_REQUESTED** [review]
Thanks for this update. We might want to wait for Ryan, as he is currently on vacation.

**Review by emilylange — CHANGES_REQUESTED** [review]
Fixes eval

```patch
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index aaf5dc9fb598..68bab658f492 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -8378,7 +8378,7 @@ with pkgs;
   })
     garage
       garage_0_7 garage_0_8
-      garage_0_7_3 garage_0_8_2;
+      garage_0_7_3 garage_0_8_3;
 
   garmin-plugin = callPackage ../applications/misc/garmin-plugin { };
 
```

**Comment by adamcstephens** [comment]
I’ve successfully upgraded my three node cluster to this version.

**Review by emilylange — APPROVED** [review]

**Comment by emilylange** [comment]
@0x4A6F any particular reason you would like to wait until ~RaitoBezarius is back from vacation?

**Comment by RaitoBezarius** [comment]
None :p ! (From a boat)

Please proceed if you deem it appropriate

**Review by 0x4A6F — APPROVED** [review]

**Comment by adamcstephens** [comment]
I verified that x86_64-darwin does actually build successfully. There are clearly no nixosTests for darwin, and looking at the aarch64-darwin build it just runs the basic package build.

It took 27 minutes to run the emulation compile on my M1 machine. The [check for this PR](https://logs.ofborg.org/?key=nixos/nixpkgs.252142&attempt_id=33d9ece1-948d-4c21-ae7f-de4cbfc433de) timed out at an hour, so maybe it was just resource contention?

**Comment by emilylange** [comment]
hydra.nixos.org managed to build garage `0.8.3` on `x86_64-darwin` as well, so all good

https://hydra.nixos.org/build/233596067

**Pull request merged**